### PR TITLE
Add singleBooking query

### PR DESCRIPTION
**Pull request by jalbertsr**

### CODE REVIEW

File: src/booking/dataloaders/SingleBooking.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import DataLoader from 'dataloader';
+import { get } from '../../common/services/HttpRequest';
+import Config from '../../../config/application';
+import { sanitizeDetail } from './ApiSanitizer';
+import type { Booking } from '../Booking';
+
+export type Args = {
+  simpleToken: string,
+  id: number,
+};
```
Comment (tbergq) [inline src/booking/dataloaders/SingleBooking.js]: could this be exact type, and read only?

File: src/booking/dataloaders/SingleBooking.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+export default function createSingleBookingLoader() {
+  return new DataLoader(
+    (bookings: $ReadOnlyArray<Args>) => batchLoad(bookings),
+    {
+      cacheKeyFn: key => parseInt(key, 10),
```
Comment (tbergq) [inline src/booking/dataloaders/SingleBooking.js]: This key is an object in this case right?
you should use `json-stable-stringify`

**Comment by mrtnzlml** [comment]
So no `viewer`? That's a pity. 😞

**Comment by tbergq** [comment]
We did not see how that would solve the problem 🤷‍♂️

**Comment by mrtnzlml** [comment]
It's just a different query design. You would go from this:

```graphql
{
  booking(bid: ID)
  singleBooking(bid: ID, simpleToken: ID)
  somethingElse(simpleToken: ID)
}
```

To this:

```graphql
{
  viewer(simpleToken: ID) {
    booking(bid: ID)
    singleBooking(bid: ID)
    somethingElse
  }
}
```

And therefore it would be clear that `booking` vs. `singleBooking` is quite unfortunate and you can simplify (maybe with `bookingViewer`):

```graphql
{
  bookingViewer(simpleToken: ID) {
    booking(bid: ID) # singleBooking not needed now
    customerBookings # could be here as well - why not, viewer can have different auth methods as well :)
  }
}
```

It was just an idea. No big deal.

**Comment by jaroslav-kubicek** [comment]
What about `simpleToken` argument to be sent in `Authorization` header. Looks like a little discrepancy to me `booking(id: ID)` with `Authorization` header vs `singleBooking(id: ID, simpleToken: string)`

**Comment by tbergq** [comment]
We still send it as a query parameter to the api-endpoint 🤷‍♂️ 

Then also, how do we separate simple_token from a full token?

File: src/booking/queries/SingleBooking.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+import GraphQLBooking from '../types/outputs/Booking';
+import type { GraphqlContextType } from '../../common/services/GraphqlContext';
+import type { BookingsItem } from '../Booking';
+
+export default {
+  type: GraphQLBooking,
+  description: 'Find booking by its id and its simple access token.',
+  args: {
+    id: {
+      type: new GraphQLNonNull(GraphQLID),
```
Comment (jaroslav-kubicek) [inline src/booking/queries/SingleBooking.js]: if it accepts only bid, then it should be `Int` - same as databaseId?

**Comment by mrtnzlml** [comment]
@jaroslav-kubicek From what I understood "simple token" is not auth method as such but more like a URL parameter. I am afraid to use `Authorization` header for this because then there is going to be another auth method and we will have to add it again.

**Comment by jaroslav-kubicek** [comment]
ok, I see, forget my note above regarding the simple token.

File: src/booking/queries/SingleBooking.js (hunk 3 above)
Comment (tbergq) [reply src/booking/queries/SingleBooking.js]: You are right, I missed that 😞 
cc @jalbertsr

**Pull request merged**